plugins/apps/threatbus_zmq_app/README.md

```diff
@@ -0,0 +1,167 @@
+Threat Bus App Plugin for ZMQ Protocol
+======================================
+
+<h4 align="center">
+
+[![PyPI Status][pypi-badge]][pypi-url]
+[![Build Status][ci-badge]][ci-url]
+[![License][license-badge]][license-url]
+
+</h4>
+
+A Threat Bus plugin that enables communication with any application that can
+communicate via [ZeroMQ].
+
+
+## Installation
+
+```sh
+pip install threatbus-vast
+```
+
+## Configuration
+
+The plugin uses ZeroMQ to communicate with applications, like the
+[VAST bridge](https://github.com/tenzir/threatbus/tree/master/apps/vast). The
+plugin serves three ZeroMQ endpoints to connect with. One endpoint for managing
+subscriptions (and thus snapshot requests). The other two endpoints are plainly
+for pub-sub operations.
+
+```yaml
+...
+plugins:
+  vast:
+    host: "127.0.0.1"
+        manage: 13370
+        pub: 13371
+        sub: 13372
+...
+```
+
+Initially, apps that want to connect with this plugin only need to know the
+`manage` endpoint. The plugin and the app negotiate all other internals for
+pub-sub message exchange at runtime. See the protocol definition below for
+details.
+
+## Management Protocol
+
+Subscriptions and unsubscriptions are referred to as `management`. All
+management messages are JSON formatted and exchanged via the `manage` ZMQ
+endpoint, that the plugin exposes.
+
+The manage endpoint uses the
+[ZeroMQ Request/Reply](https://learning-0mq-with-pyzmq.readthedocs.io/en/latest/pyzmq/patterns/client_server.html)
+pattern for message exchange. That means, all messages get an immediate response
+from the server. With each JSON reply, the zmq-app Threat Bus plugin sends a
+`status` field that indicates the success of the requested operation.
+
+### Subscribe at the Plugin
+
+To subscribe to Threat Bus via the zmq-app plugin, an app needs to send a JSON
+struct as follows to the `manage` endpoint of the plugin:
+
+```
+{
+  "action": "subscribe",
+  "topic": <TOPIC>,       # either 'threatbus/sighting' or 'threatbus/intel'
+  "snapshot": <SNAPSHOT>  # number of days for a snapshot, 0 for no snapshot
+}
+```
+In response, the app will either receive a `success` or `unsuccess` response.
+
+- Unsuccess response:
+  ```
+  {
+    "status": "unsuccess"
+  }
+  ```
+- Success response:
+  ```
+  {
+    "topic": <P2P_TOPIC>,
+    "pub_endpoint": 13371,
+    "sub_endpoint": 13372,
+    "status": "success",
+  }
+  ```
+
+  The `pub_endpoint` and `sub_endpoint` are the port numbers that are configured
```

Comment:
An endpoint usually has the form `host:port`. Why not call it `pub_port`?

Reply:
Good catch, these are indeed endpoints and not port numbers, so the `pub_endpoint` and `sub_endpoint` are exactly in the form `host:port`. I'll update the docs.